ux.symfony.com/src/Controller/ChangelogController.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Controller;
+
+use App\Service\Changelog\ChangelogProvider;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Attribute\Route;
+
+class ChangelogController extends AbstractController
+{
+    public function __construct(
+        private readonly ChangelogProvider $changeLogProvider,
+    ) {
+    }
+
+    #[Route('/changelog', name: 'app_changelog')]
+    public function __invoke(): Response
+    {
+        $changelog = $this->changeLogProvider->getChangelog();
```

Comment:
Why not also have ChangelogComponent directly?

```php
class ChangelogComponent
{
   public function __construct(private readonly ChangelogProvider $changeLogProvider,) {}

   public function getChangelog()
 {
       $this->changeLogProvider->getChangelog();
 }
}
```

Reply:
I'm not sure we want _yet_ to have routes splitted in controllers and components... but i had the same tought and that could be really usefull in "sections" like .... demos :) 

Another reason to wait a bit for the changelog is ... i'd like soon to extend it for the big packages where we have the most questions / interest (e.g. components, autocomplete, ..)

Reply:
Hmm realizing when pressing "comment" you did not suggest to create a component with route, but to have the provider into the component and not into the controller 😅 

So the answer of your actual question is: for now it's just a page., but when (it's not a distant future) we will have other parameters  ("pages" / "component" / "version") i need to check it in the controller to handle the redirections / errors / etc.